lua/cmp/config/context.lua

```diff
@@ -0,0 +1,69 @@
+local context = {}
+
+---Check if cursor is in syntax group
+---@param group string
+---@return boolean
+context.in_syntax_group = function(group)
+  local lnum, col = vim.fn.line('.'), math.min(vim.fn.col('.'), #vim.fn.getline('.'))
+  for _, syn_id in ipairs(vim.fn.synstack(lnum, col)) do
+    syn_id = vim.fn.synIDtrans(syn_id) -- Resolve :highlight links
+    if vim.fn.synIDattr(syn_id, 'name') == group then
+      return true
+    end
+  end
+end
+
+---Check if cursor is in treesitter capture
+---@param capture string
+---@return boolean
+context.in_treesitter_capture = function(capture)
+  local highlighter = require('vim.treesitter.highlighter')
+  local ts_utils = require('nvim-treesitter.ts_utils')
+  local buf = vim.api.nvim_get_current_buf()
+  local row, col = unpack(vim.api.nvim_win_get_cursor(0))
+  row = row - 1
+  if vim.api.nvim_get_mode().mode == 'i' then
+    col = col - 1
+  end
+
+  local self = highlighter.active[buf]
+  if not self then
+    return {}
```

Comment:
@max397574 Should this be false or true or maybe nil? I noticed the example use you posted checks for true specifically, but I don't know what the empty table means.

Thanks for adding this, BTW.

Reply:
I agree with groner, what is `{}` supposed to mean? Shouldn't it be `false` which is what the documentation says? @max397574